app/components/widgets/settings/radio_setting.js

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {Text, TouchableOpacity, View} from 'react-native';
+import CheckMark from 'app/components/checkmark';
+import {paddingHorizontal as padding} from 'app/components/safe_area_view/iphone_x_spacing';
+import FormattedText from 'app/components/formatted_text';
+import {makeStyleSheetFromTheme, changeOpacity} from 'app/utils/theme';
+
+export default class RadioSetting extends PureComponent {
+    static propTypes = {
+        id: PropTypes.string.isRequired,
+        label: PropTypes.node.isRequired,
+        values: PropTypes.array.isRequired,
+        value: PropTypes.string.isRequired,
+        onChange: PropTypes.func.isRequired,
+        theme: PropTypes.object.isRequired,
+        isLandscape: PropTypes.bool.isRequired,
+        helpText: PropTypes.node,
+        optional: PropTypes.bool,
+        showRequiredAsterisk: PropTypes.bool,
+    };
+
+    static defaultProps = {
+        optional: false,
+        showRequiredAsterisk: false,
+        isLandscape: false,
+    };
+
+    handleChange = (item) => {
+        this.props.onChange(this.props.id, item);
+    }
+
+    renderCheckMark = (value) => {
+        if (value === this.props.value) {
+            return (
+                <CheckMark
+                    width={12}
+                    height={12}
+                    color={this.props.theme.linkColor}
+                />
+            );
+        }
+        return null;
+    }
+
+    renderRowSeparator = (idx, separatorStyle) => {
+        const {values} = this.props;
+        if (idx === values.length - 1) {
+            return null;
+        }
+        return <View style={separatorStyle}/>;
+    }
+
+    render() {
+        const {
+            theme,
+            label,
+            helpText,
+            showRequiredAsterisk,
+            optional,
+            isLandscape,
+        } = this.props;
+        const style = getStyleSheet(theme);
+
+        let helpTextContent;
+        if (helpText) {
+            helpTextContent = (
+                <Text style={style.helpText}>
+                    {helpText}
+                </Text>
+            );
+        }
+
+        let optionalContent;
+        let asterisk;
+        if (optional) {
+            optionalContent = (
+                <FormattedText
+                    style={style.optional}
+                    id='channel_modal.optional'
```

Comment:
why are we identifying it as `channel_modal`?

Reply:
Following the same pattern as these:

https://github.com/mattermost/mattermost-mobile/blob/80ec97089ed722998829751c1b10f74cf6a29783/app/components/widgets/settings/text_setting.js#L108

https://github.com/mattermost/mattermost-mobile/blob/80ec97089ed722998829751c1b10f74cf6a29783/app/components/autocomplete_selector/autocomplete_selector.js#L141